File: Medium/2433. Find The Original Array of Prefix Xor/2433. Find The Original Array of Prefix Xor.ts
function findArray(pref: number[]): number[] {
    const n = pref.length;
    const arr: number[] = new Array(n);

    for (let i = 0; i < n; i++) {
        if (i === 0) {
            arr[i] = pref[i];
        } else {
            arr[i] = pref[i] ^ pref[i - 1];
        }
    }

    return arr;
}

File: Medium/2433. Find The Original Array of Prefix Xor/2433. Find The Original Array of Prefix Xor.rs
impl Solution {
    pub fn find_array(pref: Vec<i32>) -> Vec<i32> {
        let n = pref.len();
        let mut arr = vec![0; n];

        for i in 0..n {
            if i == 0 {
                arr[i] = pref[i];
            } else {
                arr[i] = pref[i] ^ pref[i - 1];
            }
        }

        arr
    }
}
